src/denovo_discovery/local_assembly.cpp

```diff
@@ -109,28 +125,17 @@ void get_paths_between_util(const std::string &start_kmer, const std::string &en
     auto &child_nodes = tree[start_kmer];
     auto num_children = child_nodes.size();
 
-    if (path_accumulator.length() > max_path_length) {
-        return;
-    }
-
     path_accumulator.push_back(start_kmer.back());
 
     // makes sure we get all possible cycle repitions up to the maximum length
     if (has_ending(path_accumulator, end_kmer)) {
         full_paths.push_back(path_accumulator);
-        // prevent the number of paths going above a fixed number
-        if (full_paths.size() > g_max_num_paths) {
-            BOOST_LOG_TRIVIAL(debug) << "Too many paths found " << std::to_string(full_paths.size())
-                                     << ". Skipping slice.";
-            full_paths.clear();
-            return;
-        }
     }
 
     for (unsigned int i = 0; i < num_children; ++i) {
```

Comment:
`for (const auto &child_node: child_nodes) {`
no need for `num_children` variable, right?